src/controllers/trip.controller.ts

```diff
@@ -0,0 +1,82 @@
+import { Request, Response, NextFunction } from "express";
+import {
+  mongoCreateTrip,
+  mongoDeleteTrip,
+  mongoGetTripById,
+  mongoGetTrips,
+  mongoUpdateTrip,
+} from "../services/trip.service";
+import { CustomRequest } from "../types";
+
+export const createTrip = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const trip = await mongoCreateTrip(req.body);
+    res.json(trip);
+  } catch (error) {
+    next(error);
+  }
+};
+
+export const deleteTrip = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    await mongoDeleteTrip((req as CustomRequest).user._id.toString());
+    res.json({ message: "Trip deleted successfully" });
+  } catch (error) {
+    next(error);
+  }
+};
+
+export const getTripById = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const trip = await mongoGetTripById(req.params.id);
+    res.json(trip);
+  } catch (error) {
+    next(error);
+  }
+};
+
+export const getTrips = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const { page, limit } = req.query;
```

Comment:
what happeneds if page and limit are not provided?

Reply:
return all

Reply:
are you sure? is it tested?
i know when when casting undefined to a Number() ( Number(undefined) ) it result with a NaN, woudnt it be a problem?
it will couse NaN to be sent to the mongoGetTrips()